### PR TITLE
Filter Netty native lib errors likewise the pure Java implementation

### LINKED ISSUE #3798: io.netty.channel.unix.Errors$NativeIoException: sendToAddress(..) failed: Network is unreachable
### Description

A new exception appeared in the logs after the upgrade to version `22.4.0`

Details of the exception:

class: `PeerDiscoveryAgent`

message:
```
io.netty.channel.unix.Errors$NativeIoException: sendToAddress(..) failed: Network is unreachable`

Sending to peer DiscoveryPeer{status=bonding, enode=enode://1bae6694670f0b56569c552665c3ab4c7ff2ee9e310357143d7ca5b805adae1e0dbe1b6d5db90a3de0a981f83f650fa7afa1beef6922af76688578745f4bf344@[2a01:4f9:4b:3d0a::2]:30308, firstDiscovered=1651828937464, lastContacted=0, lastSeen=0} failed, packet: 0x9050ea1980db086d76bc35f48359ced5ae4d2b409a121db7d5966f34f0dfa2ade32d1780373ac3e242e6f2a7afcb6c5f90d80360cf846088a85c9874f07ec4663ba40de118e6dac162bb3338d884dacca7f7e1fc3e20ea32f7899051a33a6f750001eb05cb8423b42bc082765f82765fd7902a0104f9004b3d0a0000000000000002827664827664846274e91f2d
```

It happened 108 in the last day, and all the logged exception seems to have in common an IPv6 address, and the cause could be that in instance running Besu has IPv6 enabled, but no IPv6 address configured, so not able to reach the network, as the exception says, in that case the error could be safely ignored and not logged at WARN level.

### Steps to Reproduce (Bug)
1. Run Besu 22.4.0 on Linux with IPv6 enabled, but no IPv6 address configured
2. Wait until Besu tries to connect to an IPv6 enode

**Expected behavior:** [What you expect to happen]
Demote the log from WARN to DEBUG, or just log at WARN once

**Actual behavior:** [What actually happens]

**Frequency:** [What percentage of the time does it occur?]

108 times in a day

### Versions (Add all that apply)
* Software version: 22.4.0
* Java version: Java11
* OS Name & Version: Amazon Linux2
* Cloud VM, type, size: AWS

### PR DESCRIPTION
**Pull request by fab-10**
Signed-off-by: Fabio Di Fabio <fabio.difabio@consensys.net>

## PR description

Since version `22.4.0` Netty prefers the native lib implementation, instead of the pure Java implementation, and this causes warn log messages in case IPv6 is not enable or configured. These error messages are intercepted and logged at debug when thrown by the pure Java implementation, but the native lib implementation triggers a different exception, so the scope of this PR is to extend the error filtering logic to intercept and log at debug the equivalent errors thrown by the native lib implementation

## Fixed Issue(s)

fixes #3798 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/p2p/build.gradle [hunk 1]
```diff
@@ -42,6 +42,8 @@ dependencies {
   implementation 'dnsjava:dnsjava'
   implementation 'io.prometheus:simpleclient'
   implementation 'io.vertx:vertx-core'
+  implementation 'io.netty:netty-transport-native-unix-common'
```
Comment (diega) [inline ethereum/p2p/build.gradle:45]: I think the order of dependencies is mostly alphabetical. If that's the case this should go above `prometheus`

**Review by diega — APPROVED** [review]
LGTM.
I just left some non blocking comments

**Review by macfarla — COMMENTED** [review]
sonar is complaining about code coverage - can we add a test?

File: ethereum/p2p/build.gradle (hunk 1 above)
Comment (fab-10) [reply ethereum/p2p/build.gradle:45]: tnx reordered both, I thought spotless would have taken care of them

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java [hunk 2]
```diff
@@ -312,38 +309,16 @@ protected void handleOutgoingPacket(final DiscoveryPeer peer, final Packet packe
...
-                      err);
-                }
+                handleOutgoingPacketErrors(err, peer, packet);
                 return;
               }
               peer.setLastContacted(System.currentTimeMillis());
             });
   }
 
+  protected abstract void handleOutgoingPacketErrors(
```
[the remark this answers is not in the document]
Comment (fab-10) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java]: right, renamed

**Comment by fab-10** [comment]
> sonar is complaining about code coverage - can we add a test?
@macfarla unfortunately there are no unit tests at all for that specific class `VertxPeerDiscoveryAgent` :disappointed:, and I think it makes sense to create tests for the entire class, not only for my minor modification, do you think we can release this, and delegate the unit tests creation to another ticket?

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java [hunk 3]
```diff
@@ -196,6 +200,53 @@ public CompletableFuture<?> stop() {
...
+          err);
+    } else if (err instanceof UnsupportedAddressTypeException) {
+      LOG.warn(
+          "Unsupported address type exception when connecting to peer {}, this is likely due to ipv6 not being enabled at runtime. "
+              + "Set logging level to TRACE to see full stacktrace",
+          peer);
+      LOG.trace(
+          "Sending to peer {} failed, packet: {}, stacktrace: {}",
+          peer,
+          wrapBuffer(packet.encode()),
```
[the remark this answers is not in the document]
Comment (fab-10) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/VertxPeerDiscoveryAgent.java]: the hope is that this error is not frequent, but yes it make sense to use lambda version for this and the debug logs

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**